### PR TITLE
Update to use new Faabric dirty tracking

### PR DESCRIPTION
**Pull request by Shillaker**
Updates for Faabric dirty tracking changes in https://github.com/faasm/faabric/pull/210

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -303,16 +303,8 @@ jobs:
...
-          grep 'Conan version' /tmp/conan-ver || exit 1
-          echo "::set-output name=conan-version::$(grep 'Conan version' /tmp/conan-ver | tr -d '[:alpha:] ')"
+          echo "::set-output name=conan-version::$(conan --version | tr -d '[:alpha:] ')"
         shell: bash
-        env:
-          FAASM_BUILD_MOUNT: /build/faasm
-          CONAN_CACHE_MOUNT_SOURCE: ~/.conan/
-      - name: Get Conan Version - Show Log
-        if: always()
-        run: cat /tmp/conan-ver-log
```
Comment (Shillaker) [inline .github/workflows/tests.yml:315]: This was broken and the caching wasn't working before, now it does AFAICT.

File: cmake/ExternalProjects.cmake [hunk 2]
```diff
@@ -98,7 +98,7 @@ add_library(AWS::s3 ALIAS aws_ext_s3_lib)
 set(FETCHCONTENT_QUIET OFF)
 FetchContent_Declare(wavm_ext
     GIT_REPOSITORY "https://github.com/faasm/WAVM.git"
-    GIT_TAG "faasm"
+    GIT_TAG "2434b3b170404b63afbd2103eecbf510dd1665f4"
```
Comment (Shillaker) [inline cmake/ExternalProjects.cmake:101]: This update is crucial and I don't trust using a branch name in `FetchContent`.

File: docs/development.md [hunk 3]
```diff
@@ -62,9 +62,6 @@ want to work on:
 
 # Faasm
 ./bin/cli.sh faasm
-
-# Faabric
-./bin/cli.sh faabric
```
Comment (Shillaker) [inline docs/development.md:67]: No longer in this repo. Done directly in Faabric checkout.

File: docs/development.md [hunk 4]
```diff
@@ -75,23 +72,18 @@ container you need to be in for each step):
 ```bash
 # --- CPP CLI ---
 # Build CPP functions and lib required for the tests
-inv func.local
-inv libfake
+inv func.local libfake
```
Comment (Shillaker) [inline docs/development.md:75]: Changed these to one-liners to be more copy-pasteable

File: src/wasm/WasmModule.cpp [hunk 5]
```diff
@@ -91,94 +92,19 @@ std::shared_ptr<faabric::util::SnapshotData> WasmModule::getSnapshotData()
...
-faabric::util::MemoryView WasmModule::getMemoryView()
+std::span<uint8_t> WasmModule::getMemoryView()
 {
     uint8_t* memBase = getMemoryBase();
     size_t currentSize = currentBrk.load(std::memory_order_acquire);
-    return faabric::util::MemoryView({ memBase, currentSize });
+    return std::span<uint8_t>(memBase, currentSize);
 }
 
-static std::string getAppSnapshotKey(const faabric::Message& msg)
```
Comment (Shillaker) [inline src/wasm/WasmModule.cpp:101]: All of this logic has been moved into the `Executor` class in Faabric now.

File: src/wavm/openmp.cpp [hunk 6]
```diff
@@ -475,107 +453,24 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
         // is just within this function group, and not the global OpenMP
         // thread number
         m.set_groupidx(i);
-        m.set_groupsize(nextLevel->numThreads);
-    }
```
Comment (Shillaker) [inline src/wavm/openmp.cpp:479]: This logic has also been absorbed into the `Executor` class in Faabric.

**Review by csegarragonz — APPROVED** [review]
LGTM. Just remember to:
* Merge CPP PR first.
* Re-tag (force) CPP code when merged.
* Merge faabric first.
* Update submodules for faabric and cpp in this PR.
* Finally merge this PR.

**Pull request merged**